server.go

```diff
@@ -33,55 +36,88 @@ func (s *Server) Listen() {
 	for {
 		conn, err := s.l.Accept()
 		if err != nil {
-			log.Fatal(err)
+			s.logf("golfish: failed to accept incomming connection: %v", err)
+			continue
 		}
 
-		go s.handleConn(conn)
+		go func() {
+			if err := s.handleConn(conn); err != nil {
+				s.logf("goldfish: unable to handle request from %v: %v", conn.RemoteAddr(), err)
+
+			}
+			if err := conn.Close(); err != nil {
+				s.logf("goldfish: failed to close connection with %v: %v", conn.RemoteAddr(), err)
+			}
+		}()
 	}
 }
 
-func (s *Server) handleConn(conn net.Conn) {
-	defer func() {
-		if err := conn.Close(); err != nil {
-			log.Printf("Failed to  close connection with client: %v", err)
-		}
-	}()
+func (s *Server) handleConn(conn io.ReadWriteCloser) error {
 	r := bufio.NewReader(conn)
+	for {
+		buf, err := s.readMessage(r)
+		if err != nil {
+			return fmt.Errorf("failed to read message from connection: %v", err)
```

Comment:
That is on purpose. You might be confused by seeing `fmt.Errorf`. But this expression returns an error, it does not print anything. 

So this line returns an error, while at lines 39, 45 and 49  errors are printed.
If the flow returns at this line, line 60, the message printed at line 45 might look like:

`goldfish: unable to handle request from 127.0.0.1:8082: failed to read message from connection: failed to peek into message: something went wrong.`